app/controllers/xapi/DocumentController.php

```diff
@@ -83,6 +83,8 @@ public function store() {
     $data['content_info'] = $this->getAttachedContent('content');
     $data['ifMatch'] = \LockerRequest::header('If-Match');
     $data['ifNoneMatch'] = \LockerRequest::header('If-None-Match');
+    $data['ifMatch'] = isset($data['ifMatch']) ? trim($data['ifMatch'], '"') : null;
+    $data['ifNoneMatch'] = isset($data['ifNoneMatch']) ? trim($data['ifNoneMatch'], '"') : null;
```

Comment:
you can't just trim these; that'll result in accepting illegal values (with extra quotes). You could either make sure to just trim the outer quotes, or a cleaner approach would be to add the quotes to the hash to make the document's ETag, then compare the ETag directly with the headers in question.